torch/utils/benchmark/utils/compile.py

```diff
@@ -0,0 +1,192 @@
+        else:
+            model(sample_input)
+        
+        end = time.time()
+
+        if sample_input.get_device() >= 0:
+            torch.cuda.synchronize()
+
+        durations.append(end - start)
+    
+    return sum(durations) / num_iters
+
+def benchmark_compile(model: torch.nn.Module, sample_input: torch.Tensor, num_iters: int = 5, backend: Optional[str] = None, mode="default", is_training=False, optimizer=None):
+    """
+    Use this utility to benchmark torch.compile
+    """
+    if backend:
+        try:
+            opt_model = torch.compile(model, backend=backend, mode=mode)
+            
+            # Compilation only happens after the first inference
+            compilation_time = bench_loop(opt_model, sample_input, 1, is_training, optimizer)
+
+        except:
+            print(f"Failed to compile {backend} with mode {mode}")
+            return None, None
+    else:
+        opt_model = model
+        compilation_time = None
+
+    # Benchmark
+    running_time = bench_loop(opt_model, sample_input, num_iters, is_training, optimizer)
+    
+    return compilation_time, running_time
+
+    
+def bench_all(model : torch.nn.Module, sample_input : torch.Tensor, num_iters : int=2, is_training : bool =False, optimizer: Optional[torch.optim.Optimizer]=None):
+    """
+    This is a simple utility that can be used to benchmark torch.compile
+    In particular it ensures that your GPU is setup to use tensor cores if it supports its
+    It also tries out all the main backends and prints a table of results so you can easily compare them all
+    Many of the backendds have their own optional dependencies so please pip install them seperately
+    
+    The tables will look like
+    Train/Inference | Backend        | Mode | Compilation Time | Average Running Time | Speedup   
+    ----------------+----------------+------+------------------+----------------------+-----------
+    Training        | Eager          | -    | -                |     0.0005           | -         
+    Training        | aot_ts_nvfuser | -    |     0.5023       |     0.0006           |    -0.3070
+    Training        | cudagraphs     | -    |     0.0242       |     0.0259           |   -51.4658
+
+    The important warnings are
+    Your model is loaded on GPU
+    Your GPU supports tensor cores, we will enable it automatically by setting `torch.set_float32_matmul_precision('high')`
+
+    If a compilation fails for any reason including the dependency not being included then we will print Failed to compile {backend} with mode {mode}
+
+
+    """
+    if next(model.parameters()).is_cuda:
+        print("Your model is loaded on GPU")
+        if torch.backends.cuda.matmul.allow_tf32 is False and torch.cuda.get_device_capability() >= (8, 0):
+            print("Your GPU supports tensor cores, we will enable it automatically by setting `torch.set_float32_matmul_precision('high')`")
+            torch.set_float32_matmul_precision("high")
```

Comment:
you never unset it, it's better to do these things in `try...finally` block so that even compilation fails or there's some other error, the initial state is restored.